interface/src/app/gene/genes.js

```diff
@@ -0,0 +1,384 @@
+      data: {
+        pageTitle: 'Gene Search'
+      }
+    })
+
+  .state('gene_network', {
+    url: '/gene_network/',
+    views: {
+      "main": {}
+    },
+    data: {
+      pageTitle: 'Gene Network'
+    }
+  });
+})
+
+.factory('SearchResults', ['$rootScope', 'Gene', function($rootScope, Gene) {
+  var queries = [];
+  var searchResults = {};
+
+  return {
+    getQueries: function() {
+      return queries;
+    },
+    getQueryResults: function(query) {
+      return searchResults[query];
+    },
+    getSearchResults: function() {
+      return searchResults;
+    },
+    remove: function(query) { // Remove a query and its associated result
+      queries = queries.filter(function(el) {
+        return el !== query;
+      });
+      delete searchResults[query];
+      $rootScope.$broadcast('results.update');
+    },
+    clear: function() { // Clear the service
+      queries = [];
+      searchResults = {};
+      $rootScope.$broadcast('results.update');
+    },
+    size: function() {
+      return queries.length;
+    },
+    search: function(qparams) {
+      // Search for genes and add the results to the service
+      Gene.search(qparams, function(data) {
+        var previousQueries = queries.length;
+        for (var i = 0; i < data.length; i++) {
+          query = data[i].search;
+          if (!searchResults[query]) {
+            //if search term didn't already exist
+            searchResults[query] = data[i]; //add it to the results
+            queries.push(query); // add to the list of queries
+          }
+        }
+        if (previousQueries !== queries.length) {
+          $rootScope.$broadcast('results.update');
+        }
+        $rootScope.$broadcast('results.searchResultsReturned');
+      });
+    }
+  };
+}])
+
+.factory('SelectedGenesFactory', function() {
+  var selected_genes = {};
+
+  return {
+    clear: function() {
+      selected_genes = {};
+    },
+
+    addGene: function(geneObj) {
+      selected_genes[geneObj.id] = geneObj;
+    },
+
+    removeGene: function(geneObj) {
+      delete selected_genes[geneObj.id];
+    },
+
+    return_genes: function() {
+      return selected_genes;
+    }
+  };
+})
+
+// Directive for whole gene search form
+.directive('geneSearchForm', ['SearchResults', function(SearchResults) {
+  return {
+    controller: ['$scope', '$rootScope', 'SearchResults',
+      function($scope, $rootScope, SearchResults) {
+
+        $scope.loadingSearchResults = false;
+
+        // Clear any existing search results
+        SearchResults.clear();
+
+        $scope.searchGenes = function() {
+          if (!$scope.genesToAdd) { //if the query is empty
+            return false;
+          }
+          $scope.loadingSearchResults = true;
+
+          $rootScope.$broadcast('results.loadingSearchResults');
+          qparams = {
+            'query': $scope.genesToAdd.query
+          };
+
+          if ($scope.organism) {
+            qparams['organism'] = $scope.organism;
```

Comment:
We will... so if this is working you could:

`//TODO: adage doesn't currently support multiple organisms
$scope.organism = 'whatever indicates human'
`